### DISCUSSION TITLE
[ZK Stack] - Running ‘zkstack ecosystem create’ - Failed reading genesis config

### DISCUSSION THREAD

**Discussion by MikelinoZk** [zkSync-Community-Hub/zksync-developers#797] [ZK Stack] [answered]
### Module Affected

- [ ] System contracts (era-system-contracts)
- [ ] Smart contracts (era-contracts)
- [ ] zkEVM (era-zk_evm)
- [ ] Server (zksync-era)
- [ ] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [X] Other (please specify below)

### Rust Version

1.75.0

### Operating System

- [ ] Ubuntu
- [X] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

From user - I wanted to test running the stack locally but, I came across an issue.
Running ‘zkstack ecosystem create’ like described [here](https://docs.zksync.io/zk-stack/running-a-zk-chain/quickstart), fails with “failed reading genesis config”.
I’ve assumed that this command will run a L1 container for me and I do not need a genesis? If not, how do I select “localhost” but with an existing local L1 network?
![Screenshot](https://github.com/user-attachments/assets/2f40503e-2c0b-4091-9a24-8c4c5b7e8461)

### Expected Behavior

See above

### Current Behavior

See above

### Prior Research

- [X] Yes, I searched for similar issues and did not find any matching ones.

**Comment by alhenaai** [comment]
It seems like you are encountering an issue with the `zkstack ecosystem create` command due to a problem with reading the genesis configuration file. This error typically occurs when the specified genesis configuration file is missing or incorrectly specified in the environment settings.

Here are a few steps you can take to resolve this issue:

1. **Check the Genesis Configuration Path**: Ensure that the path to your genesis configuration file is correctly specified and that the file exists at that location. The error message you're seeing (`failed reading genesis config`) suggests that the file might not be found.

2. **Environment Configuration**: Verify that your environment variables are set correctly, particularly those that specify paths to configuration files. This includes checking the `genesis.yaml` path in your setup.

3. **Documentation and Setup**: Revisit the [documentation](https://docs.zksync.io/zk-stack/running-a-zk-chain/quickstart) to ensure that all prerequisites and steps are followed correctly. Make sure that any step involving the genesis configuration is properly configured.

4. **Local L1 Network**: If you are trying to use an existing local L1 network, ensure that the network settings and connections are correctly configured to be recognized by the ZKsync stack setup.

If these steps do not resolve the issue, you might want to try setting up the environment from scratch to ensure all configurations are correct. For more detailed troubleshooting, you can also visit the [ZKsync developers discussions on Github](https://github.com/zkSync-Community-Hub/zksync-developers/discussions) for more in-depth support.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by bxpana** [comment]
@MikelinoZk thanks for the report, going to work on replicating this and see what's going on

**Reply by bxpana** [reply]
You are correct that a local L1 network is run for you as long as you have Docker open. What version of zkstack-cli do you have running? I tested with `0.1.0` and haven't had any issues with running it so far.

**Reply by bxpana** [reply]
If you manually moved any files that could throw off the location for the genesis.yaml but that's not completed until after running `zkstack ecosystem init --dev` If it's not working still I would try spinning up another zkstack ecosystem and chain in a new folder with the latest version of the cli with the instructions again.

**Reply by LimonChillo** [reply, marked as answer]
Hi @MikelinoZk, just found this thread here, thanks for posting this on my behalf. 
Hi @bxpana thanks for having a look into this. Yes I did run `0.1.0` too. I couldn't find the cause of this error, also switching to a new folder has produced the same. I ended up sweeping all dependencies(node,rust,foundry,...) from my machine and did a complete re-install as in https://matter-labs.github.io/zksync-era/core/latest/guides/setup-dev.html . That did it for me, now the setup works. 

Feel free to mark this as answered, thanks!

**Reply by bxpana** [reply]
@LimonChillo that's great to hear! Glad you got things running again and let us know if you run into anything else.

**Comment by dutterbutter** [comment]
Closing due to staleness please re-open if needed.
